src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -53,6 +53,8 @@ public void buildRoad(UserBean user, String gameIdString, String edgeIdString) t
         buildUtil.validateUserCanBuildRoanOnEdge(user, edgeToBuildOn);
         buildUtil.buildRoadOnEdge(user, edgeToBuildOn);
 
+        playUtil.finishGameIfTargetReached(game);
```

Comment:
it is better to keep calls of playUtil methots behind each other, so ther code will look like:


        preparationStageUtil.updateCurrentCycleBuildingNumber(game);

        playUtil.finishGameIfTargetReached(game);
        playUtil.updateAvailableUserActions(game);